usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -338,29 +382,38 @@ def construct_es_reponse(self, response) -> dict:
             row = self.append_recipient_hash_level(row)
             row.pop("parent_recipient_unique_id")
             results.append(row)
-        last_id = None
-        last_value = None
-        if len(response) > 0:
-            last_id = response[len(response) - 1].to_dict().get("generated_unique_award_id")
-            last_value = (
-                response[len(response) - 1].to_dict().get("total_loan_value")
-                if set(self.filters["award_type_codes"]) <= set(loan_type_mapping)
-                else response[len(response) - 1].to_dict().get("total_obligation")
+
+        last_record_unique_id = None
+        last_record_sort_value = None
+
+        if self.last_record_unique_id is not None:
+            has_next = len(results) > self.pagination["limit"]
+        else:
+            has_next = (
+                response.hits.total - (self.pagination["page"] - 1) * self.pagination["limit"]
+                > self.pagination["limit"]
             )
+
+        if len(response) > 0 and has_next:
+            last_record_unique_id = response[len(response) - 2].meta.sort[1]
+            last_record_sort_value = response[len(response) - 2].meta.sort[0]
+
         return {
             "limit": self.pagination["limit"],
-            "results": results,
+            "results": results[: self.pagination["limit"]],
             "page_metadata": {
                 "page": self.pagination["page"],
-                "hasNext": response.hits.total - (self.pagination["page"] - 1) * self.pagination["limit"]
-                > self.pagination["limit"],
-                "last_id": last_id,
-                "last_value": last_value,
+                "hasNext": has_next,
+                "last_record_unique_id": last_record_unique_id,
+                "last_record_sort_value": last_record_sort_value,
             },
             "messages": [get_time_period_message()],
         }
 
     def append_recipient_hash_level(self, result) -> dict:
```

Comment:
I'm fine with the change here... but I read on to the impl of this function. 
Has this always been there, or is it new for the ES impl?
Issuing an additional SQL query for each result to be returned in the response? 

Seems like a major bottleneck risk for performance. 
Maybe not so bad when paginating only 60 at a time, but if accessing this not through the website, and getting many more results ... not so good.

Reply:
Max number of results per API request is 100. While I agree this isn't optimal, the minor performance hit was documented when it was added.